### ISSUE TITLE
Delegating an action doesn't keep the `actingAs` user

### ISSUE THREAD

**Issue by alecritson** [lorisleiva/laravel-actions#57]
As the title suggests, when you delete an action within an action after using `actingAs` the user isn't passed through.

```php
(new TopLevelAction)->actingAs($user)->run();
```

```php
class TopLevelAction extends Action 
{
 public function authorise()
 {
   $this->user(); // User object
   $result = $this->delegateTo(DelegatedAction::class);
 }
}
```

```php
class TopLevelAction extends Action 
{
 public function authorise()
 {
   $this->user(); // null
 }
}
```

From a quick look at the code, perhaps something like this can be added to the extended `Action` class

```php
    /**
     * @param Action $action
     * @return static
     */
    public static function createFrom(Action $action)
    {
        return (new static)->fill($action->all())->actingAs($action->user());
    }
```

Happy to do a PR if you think this worth adding in, otherwise I can just override in my app :)

**Comment by lorisleiva** [comment]
Hi there 👋 

Well spotted, thanks!

A PR would be most welcome. 😊

**Comment by lorisleiva** [comment]
Released in `v1.1.4`
